crates/task-impls/src/consensus.rs

```diff
@@ -1397,18 +1398,15 @@ where
             // TODO do some sort of sanity check on the view number that it matches decided
         }
 
-        // let block_commitment = Some(self.block.commit());
-        if let Some(block) = &self.block {
-            let block_commitment = block.commit();
-
+        if let Some(block_commitment) = &self.block_commitment {
             let leaf = SequencingLeaf {
                 view_number: view,
                 height: parent_leaf.height + 1,
                 justify_qc: consensus.high_qc.clone(),
                 parent_commitment: parent_leaf.commit(),
                 // Use the block commitment rather than the block, so that the replica can construct
                 // the same leaf with the commitment.
-                deltas: Right(block_commitment),
+                deltas: Right(*block_commitment),
```

Comment:
Does the block header commit to the block?  Why don't we want to store the block header here like we do when we create the leaf earlier in the code (where my previous comment is)?  Will a leaf with a block commitment in it (like here) hash to the same thing as a leaf with the block header commitment in it?  Naively I would think the two would be different hashes, which might cause problems?

Reply:
Same as above. The naming was confusing. It should be `Right(*payload_commitment)`.